src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1148,7 +1148,7 @@ public final boolean holdEfficientTool(@NotNull final Block target, final BlockP
      */
     private void requestTool(@NotNull final Block target, final BlockPos pos)
     {
-        final IToolType toolType = WorkerUtil.getBestToolForBlock(target, target.getDefaultState().getBlockHardness(world, pos));
+        final IToolType toolType = WorkerUtil.getBestToolForBlock(world.getBlockState(pos), target.getDefaultState().getBlockHardness(world, pos));
```

Comment:
kept getBlockHardness here because otherwise it would require passing in the world and pos, no real benefit of doing it that way

Reply:
I don't quite follow this comment, it already has the world and pos.  Shouldn't it be using `world.getBlockState(pos)` for both (or better, as an extra line above)?

Reply:
Or better still, change the parameter from Block to BlockState, since all the callers are already doing the getBlockState anyway.

Reply:
Agreed